### PR TITLE
Remove logicsig templates from SDKs for v2.0.0 breaking changes

### PR DESCRIPTION
**Pull request by ahangsu**
From https://github.com/algorand/go-algorand-internal/issues/1415, we have already removed algorand-sdk-testing integration part for template.

We just need to remove the `template.py` file.

Part of epic https://github.com/algorand/py-algorand-sdk/issues/384.

### CODE REVIEW

**Comment by ahangsu** [comment]
good call out, I just take all of them out

**Review by algochoi — COMMENTED** [review]
Apologies for not noticing earlier:
* I think there are some errors that are only being used in templates, so we can safely delete them: `WrongContractError`, `TemplateInputError`, `TemplateError`
* There are references to the templates in the `docs` directory. I believe these are manually edited, so we can probably delete them as well?

**Comment by ahangsu** [comment]
I just removed all the unused errors and docs, maybe this applies to other deprecation PRs, we will see...

File: docs/algosdk/future/index.rst [hunk 1]
```diff
@@ -3,6 +3,5 @@ future
 .. toctree::
    :maxdepth: 10
 
-   template
```
Comment (algochoi) [inline docs/algosdk/future/index.rst:6]: I think there is also a reference in `algosdk/index.rst`: https://github.com/algorand/py-algorand-sdk/blob/develop/docs/algosdk/index.rst

Comment (ahangsu) [reply docs/algosdk/future/index.rst:6]: 👍

**Review by algochoi — APPROVED** [review]
LGTM!

**Pull request merged**